### DISCUSSION TITLE
Question about gnss ntrip client

### DISCUSSION THREAD

**Discussion by marcuzz0** [semuconsulting/pygnssutils#52] [Q&A]
I'm trying to create an ntrip client by connecting it to a ublox zed-f9p gnss receiver to send it to differential correction.
The receiver is located on the /dev/ttyACM0 serial port, how do I use the gnssntripclient suite to do this?
I tried with:

gnssntripclient --server XXX.XXX.XXX.XXX --ggamode 1 --user XXXXX --password XXXXX -P 8080 -M XXXXX

but I didn't understand how to send him the correction, could you help me?
thanks for collaboration

**Comment by semuadmin** [comment]
Hi @marcuzz0 

The gnssntripclient CLI utlity is really only intended as a testing and diagnostic tool - it simply displays NTRIP caster output at the terminal (or another designated writeable output medium); it has no facility to pass this data onto a receiver.

To do that, you'd need to build a script or application around the underlying [GNSSNtripClient](https://github.com/semuconsulting/pygnssutils/blob/b504a2b72a9764548f8ad55c5943cd234179c661/src/pygnssutils/gnssntripclient.py#L63) class. There is an example application in the /examples folder which illustrates one possible approach: [rtk_example.py](https://github.com/semuconsulting/pygnssutils/blob/main/examples/rtk_example.py). In essence, this writes the output from gnssntripclient to a queue, while a separate thread reads from this queue and passes the raw data to the receiver's serial port.

Alternatively, you could use the [PyGPSClient](https://github.com/semuconsulting/PyGPSClient) GUI application, which uses GNSSNtripClient under the covers.

Hope this helps

I'm converting this to a Discussion as it's more a question rather than an issue with the utility itself.
